### PR TITLE
New Ahelp IRC functions

### PR DESCRIPTION
**Pull request by Cyberboss**
Requested by @Shadowlight213

### CODE REVIEW

File: code/modules/admin/verbs/adminhelp.dm [hunk 1]
```diff
@@ -24,6 +24,23 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
 	QDEL_NULL(cstatclick)
 	QDEL_NULL(rstatclick)
 	return ..()
+	
+/datum/admin_help_tickets/proc/TicketByID(id)
+	var/list/lists = list(active_tickets, closed_tickets, resolved_tickets)
+	for(var/I in lists)
+		for(var/J in I)
+			var/datum/admin_help/AH = J
+			if(J.id == id)
```
Comment (lzimann) [inline code/modules/admin/verbs/adminhelp.dm]: AH and return AH

File: code/modules/admin/verbs/adminhelp.dm [hunk 2]
```diff
@@ -24,6 +24,23 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
...
+				return J
+
+/datum/admin_help_tickets/proc/TicketsByCKey(ckey)
+	. = list()
+	var/list/lists = list(active_tickets, closed_tickets, resolved_tickets)
+	for(var/I in lists)
+		for(var/J in I)
+			var/datum/admin_help/AH = J
+			if(J.initiator_ckey == ckey)
```
Comment (lzimann) [inline code/modules/admin/verbs/adminhelp.dm]: AH and return AH

**Comment by bgobandit** [comment]
can you please give more details on what this does and looks like for admins

**Comment by Cyberboss** [comment]
>!smsg cyberboss ticket list
><+N-Bot> #4, Active: #8, #27

>!smsg tacolizard ticket reopen #9 
><+N-Bot> Ticket successfully reopened

**Comment by Tacolizard** [comment]
Hey that's me

**Comment by Shadowlight213** [comment]
:+1: thanks

File: code/modules/admin/verbs/adminpm.dm [hunk 3]
```diff
@@ -223,9 +223,10 @@
 	var/compliant_msg = trim(lowertext(msg))
 	var/irc_tagged = "[sender](IRC)"
 	var/list/splits = splittext(compliant_msg, " ")
+	var/usage_string = "Usage: ticket <close|resolve|icissue|reject|reopen \[ticket #\]|list>"
```
Comment (lzimann) [inline code/modules/admin/verbs/adminpm.dm]: It's just a string but, could probably be a `/static` or define since it won't change at runtime.

Comment (RemieRichards) [inline code/modules/admin/verbs/adminpm.dm]: technically it's static already as the string table will look it up and find it.
but making it actually ```/static``` avoids a string table lookup so, still a +

File: code/modules/admin/verbs/adminpm.dm [hunk 4]
```diff
@@ -243,8 +244,36 @@
 				if(ticket)
 					ticket.Reject(irc_tagged)
 					return "Ticket #[ticket.id] successfully rejected"
+			if("reopen")
+				if(ticket)
+					return "Error: [target] already has ticket #[ticket.id] open"
+				var/fail = splits.len < 3 ? null : -1
+				if(!isnull(fail))
+					fail = text2num(splits[3])
+				if(isnull(fail))
```
Comment (lzimann) [inline code/modules/admin/verbs/adminpm.dm]: else

File: code/modules/admin/verbs/adminpm.dm (hunk 3 above)
Comment (Cyberboss) [inline code/modules/admin/verbs/adminpm.dm]: Based on the fact that static vars name collide with all other statics/globals I'd rather not

File: code/modules/admin/verbs/adminpm.dm (hunk 4 above)
Comment (Cyberboss) [inline code/modules/admin/verbs/adminpm.dm]: intentional, text2num returns null if it fails

**Pull request merged**